specs/TitleBar/titlebar-functional-spec.md

```diff
@@ -0,0 +1,479 @@
+        </StackPanel>
+    </TitleBar.Footer>
+</TitleBar>
+```
+
+## **Scenario: TitleBar with NavigationView L-Pattern Integration**
+
+<!-- TODO: Update image to have NavPane Button inline with TitleBar -->
+Titlebar and NavigationView in an L-Pattern.
+![alt text](images/titlebar-navview.png)
+
+TitleBar and NavigationView is in Minimal DisplayMode.
+Note that TitleBar IconSource is collapsed in minimal mode.  
+![alt text](images/titlebar-navview-minimal.png)
+
+XAML
+```xml
+<Grid>
+    <Grid.RowDefinitions>
+        <RowDefinition Height="Auto" />
+        <RowDefinition Height="*" />
+    </Grid.RowDefinitions>
+    <TitleBar 
+        x:Name="NavViewTitleBar" 
+        Title="NavView TitleBar"
+        Background="Transparent" 
+        IsBackButtonVisible="True"
+        IsBackButtonEnabled="{x:Bind NavFrame.CanGoBack}"
+        BackRequested="NavViewTitleBar_BackRequested"
+        PaneToggleRequested="NavViewTitleBar_PaneToggleRequested">
+        <TitleBar.IconSource>
+            <SymbolIconSource Symbol="Home"/>
+        </TitleBar.IconSource>
+    </TitleBar>
+
+    <NavigationView
+        x:Name="NavView"
+        Grid.Column="1">
+
+        <!-- TitleBar with NavigationView L-Pattern Overwriting resources -->
+        <NavigationView.Resources>
+            <!-- This is the border between NavView and NavView Content -->
+            <Thickness x:Key="NavigationViewContentGridBorderThickness">1,1,0,0</Thickness>
+            <!-- This is the rounded corner on the Top left of the L Pattern -->
+            <CornerRadius x:Key="NavigationViewContentGridCornerRadius">8,0,0,0</CornerRadius>
+        </NavigationView.Resources>
+
+        <Frame x:Name="NavFrame" />
+
+        <NavigationView.MenuItems>
+            ...
+        </NavigationView.Menuitems>
+    </NavigationView>
+</Grid>
+
+```
+
+_Note: Should the NavView resources be the default to avoid needing the extra code?_
+_This will be a question of what pattern will be shipped in WinUI Templates by default._
+
+**C#**
+```cs
+public MainWindow()
+{
+    this.InitializeComponent();
+
+    Window window = this; 
+    window.ExtendsContentIntoTitleBar = true;
+    window.SetTitleBar(this.DefaultTitleBar);
+}
+
+private void NavViewTitleBar_BackRequested(TitleBar sender, object args)
+{
+    if (NavFrame.CanGoBack)
+    {
+        NavFrame.GoBack();
+    }
+}
+
+private void NavViewTitleBar_PaneToggleRequested(TitleBar sender, object args)
+{
+    NavView.IsPaneOpen = !NavView.IsPaneOpen;
+}
+```
+
+_Note: How can we avoid this extra code-behind?_
+
+# API Pages
+
+## TitleBar class
+
+Represents a control that provides the title and system buttons for a window or application. It typically appears at the top of the window and 
+includes core functionalities such as drag, minimize, maximize, and close buttons.
+
+Namespace: Microsoft.UI.Xaml.Controls
+
+```cs
+public class TitleBar : Control
+```
+
+## TitleBar.Header property
```

Comment:
I'm surprised nobody has brought this up, after the heated discussions in https://github.com/microsoft/microsoft-ui-xaml/issues/9700. 
Header and Footer, while being concise terms, can be confusing in this context of a horizontal layout. Other terminology could be:
* Start and End (e.g. [Windows.UI.Xaml.TextAlignment.Start/End](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml?view=winrt-26100) 
TextAlignment
* Before and After (e.g. CSS, [Fluent Web UI](https://react.fluentui.dev/?path=/docs/components-input--docs#content-before-after))
* Leading and Trailing (e.g. [Apple SwiftUI Toolbar](https://developer.apple.com/documentation/swiftui/toolbaritemplacement/topbarleading))

In the end, it will be hard to objectively decide on what's best. Personally, I prefer StartElement and EndElement here. Before and After imply the center of the title bar would have the actual content, while in most apps the center of the title bar is actually empty.

Reply:
I understand the thought, but don't agree. A _head_ is not a synonym for the **top** but for the **leading** part. Animals that are "built horizontally" have their head at the front, not on top.

Reply:
I'm with Whiskhub here. Header/Footer is misleading. The only argument I see for keeping the naming is that TabView also got this wrong, but I'd rather you don't continue down the confusing naming and address it here with better names, like `Leading` and `Trailing`.

Reply:
I'm for header/footer if they're not going to change it everywhere else at the same time.